assets/training/model_management/src/azureml/model/mgmt/downloader/download_utils.py

```diff
@@ -51,18 +55,23 @@ def _download_git_model(model_uri: str, model_dir: Path) -> None:
     :param model_dir: local directory to clone model to
     :type: Path
     """
-    # do shallow fetch
-    clone_cmd = f"git clone --depth=1 {model_uri} {model_dir}"
-    exit_code, stdout = run_command(clone_cmd)
-    if exit_code != 0:
-        raise Exception(f"Could not clone repo {model_uri}. Error => {stdout}")
+    try:
+        # do shallow fetch
+        clone_cmd = f"git clone --depth=1 {model_uri} {model_dir}"
+        exit_code, stdout = run_command(clone_cmd)
+        if exit_code != 0:
+            msg = f"Could not clone repo {model_uri}. Error => {stdout}"
+            raise Exception(msg)
 
-    git_path = os.path.join(model_dir, ".git")
-    shutil.rmtree(git_path, onerror=_onerror)
-    return {
-        "download_time_utc": _get_system_time_utc(),
-        "size": _round_size(_get_size(model_dir)),
-    }
+        git_path = os.path.join(model_dir, ".git")
+        shutil.rmtree(git_path, onerror=_onerror)
+        return {
+            "download_time_utc": _get_system_time_utc(),
+            "size": _round_size(_get_size(model_dir)),
+        }
+    except Exception as e:
+        logger.error(f"Error in downloading from GIT => {e}")
```

Comment:
Let us use a custom error for this as well.

Reply:
Done